library/general/disallowedtags/template.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: k8sdisallowedtags
+  annotations:
+    description: Requires container images to have an image tag different
+      from the ones in a specified list.
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sDisallowedTags
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          properties:
+            tags:
+              type: array
+              items:
+                type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8sdisallowedtags
+
+        violation[{"msg": msg}] {
+          container := input_containers[_]
+          tags := [forbid | tag = input.parameters.tags[_] ; forbid = endswith(container.image, tag)]
```

Comment:
We should add a `:` prefix to the tag to make sure we are denying the full tag and not just a tag suffix.

For example: `myimage:shangrilatest` would be blocked under the current constraint as-written.

Reply:
Very true! I will fix this.

Reply:
Added a commit to address this. I will wait on the response on the object creation to address the other one.